src/swell/deployment/prepare_config_and_suite/prepare_config_and_suite.py

```diff
@@ -0,0 +1,601 @@
+        # sphere of suite questions but that requires some careful thought so as not to overload
+        # the user with questions.
+        suite_str = template_string_jinja2(self.logger, self.suite_str, self.experiment_dict,
+                                           True)
+
+        # 8. Build a list of tasks for each model component
+        # -------------------------------------------------
+        model_dep_tasks, all_tasks = self.get_suite_task_list_model_dep(suite_str)
+
+        # 9.1 Ask the new task questions that do not actually depend on the model
+        # -----------------------------------------------------------------------
+        for question_key in self.question_dictionary_model_ind.keys():
+
+            if 'tasks' in self.question_dictionary_model_ind[question_key].keys():
+
+                # Get list of tasks for the question
+                question_tasks = self.question_dictionary_model_ind[question_key]['tasks']
+
+                # Check whether any of model_dep_tasks are in question_tasks
+                if any(elem in question_tasks for elem in all_tasks):
+
+                    # Ask the question
+                    self.ask_a_question(self.question_dictionary_model_ind, question_key)
+
+        # 9.2 Iterate over the model_dep dictionary and ask task questions
+        # ----------------------------------------------------------------
+        for model in self.experiment_dict['model_components']:
+
+            # Iterate over the model_dep dictionary and ask questions
+            # -------------------------------------------------------
+            for question_key in self.question_dictionary_model_dep[model].keys():
+
+                # Ask only the task questions first
+                # ----------------------------------
+                if 'suites' not in self.question_dictionary_model_dep[model][question_key].keys():
+
+                    # Get list of tasks for the question
+                    question_tasks = \
+                        self.question_dictionary_model_dep[model][question_key]['tasks']
+
+                    # Check whether any of model_dep_tasks are in question_tasks
+                    if any(elem in question_tasks for elem in model_dep_tasks[model]):
+
+                        # Ask the question
+                        self.ask_a_question(self.question_dictionary_model_dep[model], question_key,
+                                            model)
+
+        # Return the main experiment dictionary
+        return self.experiment_dict, self.questions_dict
+
+    # ----------------------------------------------------------------------------------------------
+
+    def ask_a_question(self, full_question_dictionary, question_key, model=None):
+
+        # Set flag for whether the question should be asked
+        ask_question = True
+
+        # Has the question already been asked?
+        if question_key in self.experiment_dict.keys():
```

Comment:
`"x" in somedict` automatically checks for presence of keys, and is much faster (uses dict hashing, so O(1) performance, whereas searching the list `somedict.keys()` is O(n). Here, it doesn't matter, but get into the habit!

```suggestion
        if question_key in self.experiment_dict:
```